resources/js/components/felte-form/FelteForm.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Form,
+  FormConfigWithoutTransformFn,
+  KnownHelpers,
+  Obj,
+  Paths,
+} from "@felte/core";
+import { reporter } from "@felte/reporter-solid";
+import { createForm } from "@felte/solid";
+import { type KnownStores } from "@felte/solid/dist/esm/create-accessor";
+import { validator } from "@felte/validator-zod";
+import { useTransContext } from "@mbarzda/solid-i18next";
+import { isAxiosError } from "axios";
+import { Api } from "data-access/memo-api/types";
+import { Context, JSX, createContext, splitProps, useContext } from "solid-js";
+import toast from "solid-toast";
+import { ZodSchema } from "zod";
+
+type FormContextValue<T extends Obj = Obj> = {
+  props: FormProps<T>;
+  form: Form<T> & KnownHelpers<T, Paths<T>> & KnownStores<T>;
+};
+
+const FormContext = createContext(undefined, {
+  name: "FormContext",
+});
+
+type FormProps<T extends Obj = Obj> = Omit<
+  JSX.FormHTMLAttributes<HTMLFormElement>,
+  "onSubmit" | "onError"
+> &
+  FormConfigWithoutTransformFn<T> & {
+    schema: ZodSchema<T>;
+  };
+
+/**
+ * Wrapper for felte's `createForm`
+ *
+ * Includes solidjs' Provider, that stores
+ * createForm's data and component props
+ */
+export const FelteForm = <T extends Obj = Obj>(props: FormProps<T>) => {
+  const [t] = useTransContext();
+  const [local, createFormOptions, formProps] = splitProps(
+    props,
+    ["children", "schema"],
+    [
+      "debounced",
+      "extend",
+      "initialValues",
+      "onError",
+      "onSubmit",
+      "onSuccess",
+      "transform",
+      "validate",
+      "warn",
+    ]
+  );
+
+  const form = createForm<T>({
+    ...createFormOptions,
+    extend: [validator({ schema: local.schema }), reporter],
+    onError: (error, ctx) => {
+      createFormOptions?.onError?.(error, ctx);
+      if (isAxiosError<Api.ErrorResponse>(error)) {
+        error.response?.data.errors.forEach((error) => {
+          if (Api.isValidationError(error)) {
+            ctx.setErrors(
+              error.field,
+              // @ts-expect-error setErrors doesn't work properly with generic type
+              t(error.code, {
```

Comment:
Wyciągnij wynik tego do stałej a potem może być nawet cast do any jak o mnie chodzi. Ale żeby było widać że chociaż to t() nie jest podejrzane.

Reply:
poprawione